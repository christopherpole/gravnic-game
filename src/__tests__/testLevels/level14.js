import { MOVE_UP, MOVE_DOWN, ENTITIES, levelIsComplete, makeMoves } from '../../index';

describe('Level 14', () => {
  it('Matches the correct snapshot', () => {
    const gameState = [
      [
        {
          staticEntity: { id: 2, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#FFFF00', id: 1 },
        },
      ],
      [{ staticEntity: { id: 3, entityId: ENTITIES.BLACK_HOLE }, movableEntity: null }],
      [{ staticEntity: { id: 4, entityId: ENTITIES.LAVA }, movableEntity: null }],
      [{ staticEntity: { id: 5, entityId: ENTITIES.BLACK_HOLE }, movableEntity: null }],
      [
        {
          staticEntity: { id: 7, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#ff0000', id: 6 },
        },
      ],
      [
        {
          staticEntity: { id: 9, entityId: ENTITIES.FLOOR },
          movableEntity: { entityId: ENTITIES.BLOCK, color: '#008000', id: 8 },
        },
      ],
    ];

    const gameStates = makeMoves(gameState, [MOVE_UP, MOVE_DOWN]);
    expect(gameStates).toMatchSnapshot();
    expect(
      levelIsComplete(
        gameStates[gameStates.length - 1][gameStates[gameStates.length - 1].length - 1],
      ),
    ).toBe(true);
  });
});
