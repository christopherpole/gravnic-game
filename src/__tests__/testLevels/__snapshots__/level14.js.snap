// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level 14 Matches the correct snapshot 1`] = `
Array [
  Array [
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#ff0000",
            "entityId": "BLOCK",
            "id": 6,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 5,
            "shrinking": true,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 8,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#008000",
            "entityId": "BLOCK",
            "id": 8,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
            "shrinking": false,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
          },
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
            "shrinking": false,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": Object {
            "color": "#FFFF00",
            "entityId": "BLOCK",
            "id": 1,
            "shrinking": true,
          },
          "staticEntity": Object {
            "entityId": "BLACK_HOLE",
            "id": 3,
            "shrinking": true,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
            "shrinking": false,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
    Array [
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 2,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 3,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "LAVA",
            "id": 4,
            "shrinking": false,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 5,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 7,
          },
        },
      ],
      Array [
        Object {
          "movableEntity": null,
          "staticEntity": Object {
            "entityId": "FLOOR",
            "id": 9,
          },
        },
      ],
    ],
  ],
]
`;
